lib/iex/lib/iex/introspection.ex

```diff
@@ -677,16 +683,24 @@ defmodule IEx.Introspection do
         no_beam(module)
 
       {:ok, types} ->
-        printed =
-          for {kind, {^type, _, args}} = typespec <- types,
-              kind != :typep,
-              length(args) == arity do
-            type_doc(module, type, arity, typespec)
-            |> print_typespec()
-          end
-
-        if printed == [] do
-          types_not_found_or_private("#{inspect(module)}.#{type}")
+        types
+        |> Enum.filter(
+          &match?(
+            {kind, {^type, _, args}} when kind in [:type, :opaque] and length(args) == arity,
+            &1
+          )
+        )
+        |> Enum.sort_by(fn {_, {name, _, args}} -> {name, length(args)} end)
```

Comment:
I think you cannot have more than one result here for `{module, type, arity}`, @josevalim?

Reply:
@fertapric Adressed in 4250b33c872169fc5b4d57264b22c83b1701a3ab

Reply:
The `Enum.sort_by` can also be removed

Reply:
@fertapric Adressed in 1a6c82f5685b771a9766aafca9215bb39d37c6df